pkg/auth/providers/common/usermanager.go

```diff
@@ -149,6 +149,14 @@ func (m *userManager) SetPrincipalOnCurrentUserByUserID(userID string, principal
 		return nil, err
 	}
 
+	// ensure this principal is unique to this user
+	if conflict, err := m.GetUserByPrincipalID(principal.Name); err != nil {
+		return nil, err
+	} else if conflict != nil && conflict.UID != user.UID {
+		logrus.Errorf("Cannot set principal \"%s\" on user \"%s\". Principal already in use on user \"%s\".", principal.Name, user.UID, conflict.UID)
```

Comment:
Not sure if this is the format in here, in RKE we usually use `[%s]` for variables.

Reply:
I like that better ill switch